code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/BasicPushTemplate.kt

```diff
@@ -12,10 +12,77 @@
 package com.adobe.marketing.mobile.services.ui.notification.templates
 
 import android.content.Intent
+import androidx.annotation.VisibleForTesting
+import com.adobe.marketing.mobile.services.Log
 import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
 import com.adobe.marketing.mobile.util.DataReader
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
 
 internal class BasicPushTemplate : AEPPushTemplate {
+
+    private val SELF_TAG = "BasicPushTemplate"
+
+    /** Class representing the action button with label, link and type  */
+    class ActionButton(val label: String, val link: String?, type: String?) {
+        val type: PushTemplateConstants.ActionType
+
+        init {
+            this.type = PushTemplateConstants.ActionType.valueOf(type ?: PushTemplateConstants.ActionType.NONE.name)
```

Comment:
Can you wrap this in a try/catch to swallow `IllegalArgumentException` or `Exception` in case of an invalid `type` string